packages/schemas/src/types/log.ts

```diff
@@ -1,173 +1,223 @@
+import { z } from 'zod';
+
 import { Log } from '../db-entries';
 
 export enum LogResult {
   Success = 'Success',
   Error = 'Error',
 }
 
-export type BaseLogPayload = {
-  result?: LogResult;
-  error?: Record<string, unknown>;
-  ip?: string;
-  userAgent?: string;
-  applicationId?: string;
-  sessionId?: string;
-};
-
-type ArbitraryLogPayload = Record<string, unknown>;
-
-type RegisterUsernamePasswordLogPayload = ArbitraryLogPayload & {
-  userId?: string;
-  username?: string;
-};
-
-type RegisterEmailSendPasscodeLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  connectorId?: string;
-};
-
-type RegisterEmailLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  code?: string;
-  userId?: string;
-};
-
-type RegisterSmsSendPasscodeLogPayload = {
-  phone?: string;
-  connectorId?: string;
-} & ArbitraryLogPayload;
-
-type RegisterSmsLogPayload = ArbitraryLogPayload & {
-  phone?: string;
-  code?: string;
-  userId?: string;
-};
-
-type RegisterSocialBindLogPayload = ArbitraryLogPayload & {
-  connectorId?: string;
-  userInfo?: object;
-  userId?: string;
-};
-
-type RegisterSocialLogPayload = RegisterSocialBindLogPayload & {
-  code?: string;
-  state?: string;
-  redirectUri?: string;
-  redirectTo?: string;
-};
-
-type SignInUsernamePasswordLogPayload = ArbitraryLogPayload & {
-  userId?: string;
-  username?: string;
-};
-
-type SignInEmailSendPasscodeLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  connectorId?: string;
-};
-
-type SignInEmailLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  code?: string;
-  userId?: string;
-};
-
-type SignInSmsSendPasscodeLogPayload = ArbitraryLogPayload & {
-  phone?: string;
-  connectorId?: string;
-};
-
-type SignInSmsLogPayload = ArbitraryLogPayload & {
-  phone?: string;
-  code?: string;
-  userId?: string;
-};
-
-type SignInSocialBindLogPayload = ArbitraryLogPayload & {
-  connectorId?: string;
-  userInfo?: object;
-  userId?: string;
-};
-
-type SignInSocialLogPayload = SignInSocialBindLogPayload & {
-  code?: string;
-  state?: string;
-  redirectUri?: string;
-  redirectTo?: string;
-};
-
-type ForgotPasswordSmsSendPasscodeLogPayload = ArbitraryLogPayload & {
-  phone?: string;
-  connectorId?: string;
-};
-
-type ForgotPasswordSmsLogPayload = ArbitraryLogPayload & {
-  phone?: string;
-  code?: string;
-  userId?: string;
-};
-
-type ForgotPasswordEmailSendPasscodeLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  connectorId?: string;
-};
-
-type ForgotPasswordEmailLogPayload = ArbitraryLogPayload & {
-  email?: string;
-  code?: string;
-  userId?: string;
-};
-
-type ForgotPasswordResetLogPayload = ArbitraryLogPayload & {
-  userId?: string;
-};
+export const logResultGuard = z.nativeEnum(LogResult);
+
+export const baseLogPayloadGuard = z.object({
```

Comment:
Just for curiosity, what is the reason to change all those types to zod.

Reply:
I want to get a valid literal log type by passing in `flow`(sign-in | register | forgot-password), `operation` (send | verify) and media (sms | email). Try to get the list of log types to build a guard that can infer the right type of the literal string.